jetstream/message.go

```diff
@@ -27,55 +27,87 @@ import (
-		// Reply returns a reply subject for a message
+
+		// Reply returns a reply subject for a message.
 		Reply() string
 
-		// Ack acknowledges a message
-		// This tells the server that the message was successfully processed and it can move on to the next message
+		// Ack acknowledges a message This tells the server that the message was
+		// successfully processed and it can move on to the next message.
 		Ack() error
-		// DoubleAck acknowledges a message and waits for ack from server
+
+		// DoubleAck acknowledges a message and waits for ack reply from the server.
+		// While it impacts performance, it is useful for scenarios where
+		// message loss is not acceptable.
 		DoubleAck(context.Context) error
-		// Nak negatively acknowledges a message
-		// This tells the server to redeliver the message
+
+		// Nak negatively acknowledges a message. This tells the server to
+		// redeliver the message.
 		Nak() error
-		// NakWithDelay negatively acknowledges a message
-		// This tells the server to redeliver the message
-		// after the given `delay` duration
+
+		// NakWithDelay negatively acknowledges a message. This tells the server
+		// to redeliver the message after the given delay.
 		NakWithDelay(delay time.Duration) error
-		// InProgress tells the server that this message is being worked on
-		// It resets the redelivery timer on the server
+
+		// InProgress tells the server that this message is being worked on. It
+		// resets the redelivery timer on the server.
 		InProgress() error
-		// Term tells the server to not redeliver this message, regardless of the value of nats.MaxDeliver
+
+		// Term tells the server to not redeliver this message, regardless of
+		// the value of MaxDeliver.
 		Term() error
 	}
 
 	// MsgMetadata is the JetStream metadata associated with received messages.
 	MsgMetadata struct {
-		Sequence     SequencePair
+		// Sequence is the sequence information for the message.
+		Sequence SequencePair
+
+		// NumDelivered is the number of times this message was delivered to the
+		// consumer.
 		NumDelivered uint64
-		NumPending   uint64
-		Timestamp    time.Time
-		Stream       string
-		Consumer     string
-		Domain       string
+
+		// NumPending is the number of messages pending for this consumer.
+		NumPending uint64
+
+		// Timestamp is the time the message was received.
+		Timestamp time.Time
+
+		// Stream is the stream name this message is stored on.
+		Stream string
+
+		// Consumer is the consumer name this message was delivered to.
+		Consumer string
+
+		// Domain is the domain this message was received on.
+		Domain string
 	}
 
-	// SequencePair includes the consumer and stream sequence info from a JetStream consumer.
+	// SequencePair includes the consumer and stream sequence numbers for a
+	// message.
 	SequencePair struct {
+		// Consumer is the consumer sequence number for a message. This is the
+		// total number of messages the consumer has seen (including
+		// redeliveries).
```

Comment:
Yes, this is what I meant by `(including redeliveries)` - I'll try to phrase it better

Reply:
Changed to:
```go
		// Consumer is the consumer sequence number for message deliveries. This
		// is the total number of messages the consumer has seen (including
		// redeliveries).
```

Not really sure about this...

Reply:
Hmm, yeah kind of right I guess, I think we dont know how this will change when we change consumer raft details, might be worth being vague indicating its the consumer transaction sequence and does not relate to stream sequence rather than try to say its a specific thing?